### PR TITLE
fix(ListViewExport) lisView does not support exporting fields from other modules

### PR DESCRIPTION
**Pull request by maliknajjar**
* fixed the issue by using the lisview generated query instead creating it again
* supported features listed in the second image
* adding a notice message if the user used the search
* removed the code that is not used any more
image
image

### CODE REVIEW

File: data/CRMEntity.php [hunk 1]
```diff
@@ -3758,7 +3683,7 @@ public function getOrderBy() {
 		$order_by = '';
 		$customView = new CustomView($cmodule);
 		$viewid = $customView->getViewId($cmodule);
-		$sortfieldbyfirst = cbCVManagement::getFieldValuesByCvId($viewid)['sortfieldbyfirst'];
```
Comment (joebordes) [inline data/CRMEntity.php:3761]: I fixed this warning a few days ago like this:  https://github.com/tsolucio/corebos/commit/2e0a1cbf43b71fed827c8bf90e517d25883a3552   which is correct/better?

File: include/utils/export.php [hunk 2]
```diff
@@ -97,115 +97,58 @@ function export($type, $format = 'CSV') {
...
-			$query .= " and $focus->table_name.$focus->table_index in (" . generateQuestionMarks($idstring) . ')';
-			$params[] = $idstring;
-			// END
+	if ($export_data == 'currentpage') {
+		$query .= $_SESSION['limitQuery'];
+	} elseif ($export_data == 'selecteddata') {
+		$idsArray = explode(';', vtlib_purify($_REQUEST['idstring']));
+		$where = '';
+		foreach ($idsArray as $key => $value) {
+			if (!empty($value)) {
```
Comment (joebordes) [inline include/utils/export.php:112]: why isnt this an `in ('. generateQuestionMarks($idstring) .')` ?  better as a list of ORs?

File: modules/com_vtiger_workflow/com_vtiger_workflow.php [hunk 3]
```diff
@@ -25,62 +25,6 @@ class com_vtiger_workflow extends CRMEntity {
 	public function track_view($user_id, $current_module, $id = '') {
 	}
 
-	public function create_export_query($where) {
```
Comment (joebordes) [inline modules/com_vtiger_workflow/com_vtiger_workflow.php:28]: does this work?

File: include/utils/export.php [hunk 4]
```diff
@@ -105,6 +105,9 @@ function export($type, $format = 'CSV') {
 
 	if (method_exists($focus, 'create_export_query')) {
 		$query = $focus->create_export_query('');
+		global $log;
+		$log->fatal('PPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPP');
+		$log->fatal($query);
```
Comment (joebordes) [inline include/utils/export.php]: :-)

Comment (maliknajjar) [reply include/utils/export.php]: I removed them 👍

**Pull request closed without merging**